clayui.com/src/components/IconSearch/IconSearch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+import icons from './icons.json';
+
+const spritemap = '/images/icons/icons.svg';
+
+export default () => {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [filteredIconNames, setFilteredIconNames] = useState([]);
+
+	const handleIconSearch = event => {
+		setSearchQuery(event.target.value);
+	};
+
+	useEffect(() => {
+		const filteredIcons = icons.filter(icon => {
+			if (icon.aliases.includes(searchQuery)) {
+				return icon.name;
+			}
+		});
+
+		setFilteredIconNames(
+			filteredIcons.map(icon => {
+				return icon.name;
+			})
+		);
+	}, [searchQuery]);
+
+	return (
+		<>
+			<ClayForm.Group>
+				<label htmlFor="iconSearch">{'Search Icons'}</label>
+
+				<ClayInput
+					id="iconSearch"
+					onChange={handleIconSearch}
+					placeholder="Search icons"
+					type="text"
+					value={searchQuery}
+				/>
+			</ClayForm.Group>
+
+			<ul className="lexicon-icon-list list-unstyled">
+				{filteredIconNames.length
```

Comment:
We can simplify this block a bit by doing something like this...

```js
// This would live above the jsx
const list = filteredIconNames.length ? filteredIconNames : icons;

// This would live where the list is rendered
{list.map(icon => (
	<li key={icon.name}>
		<ClayIcon
			spritemap={spritemap}
			symbol={icon.name}
		/>

		<span>{icon.name}</span>
	</li>
))}
```

Also note that we should avoid using the index for the `key`. Sometimes its fine to use, but if we can use a unique identifier instead, it will re-render more accurately.

Reply:
Ah yea, good catch. I didn't go deep into cleanup since I just wanted to get the filtering to work and for it to actually render somewhere. Thanks for the in-depth comments.